src/test/test-nss.c

```diff
@@ -450,5 +451,117 @@ int main(int argc, char **argv) {
                 log_info(" ");
         }
 
+        return 0;
+}
+
+static char** arg_modules = NULL;
+static char** arg_names = NULL;
+
+static int parse_argv(int argc, char **argv, struct local_address **addresses, int *n_addresses) {
+        int r, i, n = 0;
+        _cleanup_strv_free_ char **names = NULL;
+        _cleanup_free_ struct local_address *addrs = NULL;
+        size_t n_allocated = 0;
+
+        if (argc > 1) {
+                r = strv_push(&arg_modules, argv[1]);
+                if (r < 0)
+                        return r;
+        }
+
+        for (i = 2; i < argc; i++) {
+                r = strv_push(&names, argv[i]);
+                if (r < 0)
+                        return r;
+        }
+
+        if (!arg_modules) {
+                arg_modules = strv_split_nulstr(MODULE1 MODULE2 MODULE3 MODULE4);
+                if (!arg_modules)
+                        return log_oom();
+        }
+
+        if (names) {
+                char **name;
+                int family;
+                union in_addr_union address;
+
+                STRV_FOREACH(name, names) {
+                        r = in_addr_from_string_auto(*name, &family, &address);
+                        if (r < 0 && r != -EINVAL)
+                                return log_error_errno(r, "Failed to parse address \"%s\": %m", *name);
+                        if (r >= 0) {
+                                char *x;
+
+                                if (!GREEDY_REALLOC0(addrs, n_allocated, n + 1))
+                                        return -ENOMEM;
+
+                                addrs[n++] = (struct local_address) { .family = family,
+                                                                      .address = address };
+
+                                /* use a placeholder to mark items for deletion later on */
+                                x = strdup("__ignore__");
+                                if (!x)
+                                        return log_oom();
+
+                                *name = x;
+                        }
+                }
+        } else {
+                char *hostname;
+
+                names = strv_split_nulstr("localhost\0" "gateway\0" "foo_no_such_host\0");
```

Comment:
not that it mattrers, but there's also `strv_new("localhost", "gateway", "foo_no_such_host", NULL)` which might be a bit more readable.